repose-aggregator/core/repose-core/src/main/scala/org/openrepose/powerfilter/ReposeFilter.scala

```diff
@@ -19,7 +19,193 @@
+import org.openrepose.commons.utils.opentracing.ScopeHelper.{closeSpan, startSpan}
+import org.openrepose.commons.utils.scala.TryWith
+import org.openrepose.commons.utils.servlet.http.ResponseMode.MUTABLE
+import org.openrepose.commons.utils.servlet.http.{HttpServletRequestWrapper, HttpServletResponseWrapper}
+import org.openrepose.core.services.reporting.metrics.MetricsService
+import org.openrepose.core.services.uriredaction.UriRedactionService
+import org.openrepose.core.spring.ReposeSpringProperties
+import org.openrepose.core.systemmodel.config.{Filter => _}
+import org.openrepose.nodeservice.containerconfiguration.ContainerConfigurationService
+import org.openrepose.nodeservice.response.ResponseHeaderService
+import org.openrepose.powerfilter.ReposeFilter._
+import org.slf4j.{LoggerFactory, MDC}
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.web.filter.DelegatingFilterProxy
+
+import scala.util.{Failure, Success}
+
+@Named("reposeFilter")
+class ReposeFilter @Inject()(@Value(ReposeSpringProperties.NODE.NODE_ID) nodeId: String,
+                             @Value(ReposeSpringProperties.CORE.REPOSE_VERSION) reposeVersion: String,
+                             optMetricsService: Optional[MetricsService],
+                             reposeFilterLoader: ReposeFilterLoader,
+                             containerConfigurationService: ContainerConfigurationService,
+                             tracer: Tracer,
+                             uriRedactionService: UriRedactionService,
+                             responseHeaderService: ResponseHeaderService
+                            )
+  extends DelegatingFilterProxy {
+
+  private val optMetricRegistry: Option[MetricRegistry] = Option(optMetricsService.orElse(null)).map(_.getRegistry)
+
+  override def initFilterBean(): Unit = {
+    ThisLogger.info("{} -- Initializing ReposeFilter", nodeId)
+    reposeFilterLoader.setServletContext(getServletContext)
+    ThisLogger.trace("{} -- initialized.", nodeId)
+  }
+
+  override def doFilter(request: ServletRequest, response: ServletResponse, chain: FilterChain): Unit = {
+    reposeFilterLoader.getFilterContextList match {
+      case Some(filterContextList) =>
+        ThisLogger.trace("ReposeFilter processing request...")
+        val startTime = System.currentTimeMillis
+        if (Option(request.asInstanceOf[HttpServletRequest].getHeader(TRACE_REQUEST)).isDefined) {
+          MDC.put(TRACE_REQUEST, "true")
+        }
+
+        val contentBodyReadLimit = containerConfigurationService.getContentBodyReadLimit
+        val requestBodyInputStream =
+          if (contentBodyReadLimit.isPresent) {
+            new LimitedReadInputStream(contentBodyReadLimit.get, request.getInputStream)
+          } else {
+            request.getInputStream
+          }
+
+        val wrappedResponse = new HttpServletResponseWrapper(response.asInstanceOf[HttpServletResponse], MUTABLE, MUTABLE)
```

Comment:
With the removal of the RMS are these modes still correct?

Reply:
These need to remain `MUTABLE` since we are using the new `force` option of the wrapper's `sendError` method.